llvm/include/llvm/IR/DiagnosticInfo.h

```diff
@@ -139,7 +139,7 @@ class DiagnosticInfo {
 using DiagnosticHandlerFunction = std::function<void(const DiagnosticInfo &)>;
 
 class DiagnosticInfoGeneric : public DiagnosticInfo {
-  const Twine &MsgStr;
+  const Twine MsgStr;
```

Comment:
All of the Twine usage in Diagnostic info is super sketchy. I think we can drop this from this change and do it more comprehensively here: https://github.com/llvm/llvm-project/pull/136371